public/app/features/dashboard/state/analyticsProcessor.ts

```diff
@@ -0,0 +1,56 @@
+import { getDashboardSrv } from '../services/DashboardSrv';
+
+import { PanelData, LoadingState, DataSourceApi } from '@grafana/data';
+
+import { reportMetaAnalytics, MetaAnalyticsEventPayload } from '@grafana/runtime';
+
+export function getAnalyticsProcessor(datasource: DataSourceApi) {
+  let done = false;
+
+  return (data: PanelData) => {
+    if (!data.request || done) {
+      return;
+    }
+
+    if (data.state !== LoadingState.Done && data.state !== LoadingState.Error) {
+      return;
+    }
+
+    const eventData: MetaAnalyticsEventPayload = {
+      datasourceName: datasource.name,
+      datasourceId: datasource.id,
+      panelId: data.request.panelId,
+      dashboardId: data.request.dashboardId,
+      // app: 'dashboard',
+      // count: 1,
+      dataSize: 0,
+      duration: data.request.endTime - data.request.startTime,
+      eventName: 'data-request',
+      // sessionId: '',
+    };
+
+    // enrich with dashboard info
+    const dashboard = getDashboardSrv().getCurrent();
+    if (dashboard) {
+      eventData.dashboardId = dashboard.id;
+      eventData.dashboardName = dashboard.title;
+      eventData.dashboardUid = dashboard.uid;
+      eventData.folderName = dashboard.meta.folderTitle;
+    }
+
+    if (data.series.length > 0) {
+      // estimate size
+      eventData.dataSize = data.series.length * data.series[0].length;
+    }
+
+    if (data.error) {
+      eventData.error = data.error.message;
+    }
+
+    reportMetaAnalytics(eventData);
```

Comment:
@sakjur `reportMetaAnalytics` creates `MetaAnalyticsEvent` which is consumed by `MetaAnalyticsBackend` (https://github.com/grafana/grafana-enterprise/pull/305)